Add IsDown Integration (#1766)
DataDog/integrations-extras: isdown/README.md

# Agent Check: IsDown

## Overview

[IsDown][1] is a status page aggregator and outage monitoring tool that helps businesses monitor their dependencies. You can provide your team real-time monitoring and instant notifications for outages in all your tools and cloud providers. IsDown monitors more than 2000 status pages.

With this integration, you can receive alerts from third-party dependencies in Datadog, monitor business critical services, and understand the frequency of outages all within the out-of-the-box dashboard.

## Setup

### IsDown

To connect Datadog to IsDown, you need to generate an [API key][5] in Datadog and add it to your IsDown account.

1. Use your existing account or create a new one in [IsDown][1].
2. Log in to your account and go to the **Notifications** page.
3. Add your API Key and select if you're using the US or EU [Datadog site][4].
4. Click **Save**.
5. Select the services you want to monitor.
6. In each service you can choose the notification settings.

### Uninstallation

1. Go to the **Notifications** page in IsDown.
2. Unselect Datadog and click **Save**.

> 3. In Datadog, delete your API Key dedicated to IsDown.

Reviewer: This should be updated if your integration is using OAuth.

Author: Hi @eho1307. I think this stays the same, or I'm supposed to delete the API Key from the application side?

Reviewer: You can instead say "Ensure that all API keys associated with this integration have been disabled by searching for the integration name on the [API Keys management page](/organization-settings/api-keys?filter=) in Datadog."

Author: Sounds good! Thank you.

## Data Collected

### Service Checks

See [service_checks.json][3] for a list of service checks provided by this integration.

### Events

IsDown sends events for each outage that happens in the services you monitor. It sends two types of events, one for the start of the outage and one for the end of the outage. The events are sent with the following attributes:
- Title: The name of the service with the outage.

> - Device Name: The name of the service.

Reviewer 2: Should this line read `The name of the device.`? It looks like the service name is already used in `Title`.

Author: Thank you for pointing that out. I've removed the line and will not send that info since it's duplicated.

## Setup

1. Use your existing account or create a new one in [IsDown][1].
2. Log in to your account and go to the **Notifications** page.

> 3. Click on the checkbox to select Datadog and then click 'Connect to Datadog'.

Reviewer 2: ```suggestion
3. Click on the checkbox to select Datadog and then click **Connect to Datadog**.
```

> 4. You will be redirected to Datadog to authorize the application. IsDown will create an API key that will only have access to what IsDown needs. Send events and service checks to Datadog.

Reviewer 2: ```suggestion
4. You are then redirected to Datadog to authorize the application. IsDown creates an API key that only has access to what IsDown needs to send events and service checks to Datadog.
```

Reviewer 2: Let me know if the changes here introduce any inaccuracies.

> 5. After autorization you will be redirected to IsDown.

Reviewer 2: ```suggestion
5. After authorization you are redirected to IsDown.
```

> 6. Select the services you want to monitor.

Reviewer 2: ```suggestion
6. Select the services to monitor.
```

> 7. In each service you can choose the notification settings.

Reviewer 2: ```suggestion
7. Configure any desired notification settings for each service.
```
